### ISSUE TITLE
`preventMouse` to suppress form submits

### ISSUE THREAD

**Issue by dominikstraessle** [xcash/bootstrap-autocomplete#65]
**Is your feature request related to a problem? Please describe.**
I'm using this library for forms within a ruby on rails application. So the user can type a name in the form field and select the matching name form the autocomplete dropdown and it fills in the name. This works well as long as the Enter-Key is used (`preventEnter` is set to true). But when I click on a result with the mouse, the form is submitted/reloaded. This is not the wanted/expected behaviour.

**Describe the solution you'd like**  
I want to configure a `preventMouse` option. This makes the results clickable without submiting the form.

Another possible solution would be to not use `<a>`-tags but buttons with click-listeners inside the dropdown.

**Comment by klausmeyer** [comment]
Having the same problem in similar context.
That option would be really great I agree.

I tried patching the code like this which worked fine:

```diff
- .attr('href', '#!')
+ .attr('href', 'javascript:void();')
```

Maybe this could be the change triggered by that new config option?

I'd love to help implementing this but my JavaScript experience is very rusty and I never worked with typescript and npm build toolchain before 😁 

For now I'll go with in my local JavaScript files:

```js
$(document).on('click', '.dropdown-item', function(e) { e.preventDefault; return false; })
```

**Comment by xcash** [comment]
It's fixed in last dev release as it was related to #76 
Will be fixed in 2.3.6.
